commands/render.js

```diff
@@ -143,31 +135,29 @@ function renderFrames(records, options) {
       }
 
       progressBar.tick();
-      if (progressBar.complete) {
-        console.log('Finished rendering frames in ' + (Date.now() - start) + 'ms.');
-        resolve();
-      }
     }
 
-    // React on rendering errors observed on its stderr
+    // Track rendering errors observed on stderr
     function onError(error) {
 
-      // First kill the rendering process
-      render.kill();
-
-      // Check for GPU error
-      if (!!error && error instanceof Buffer && error.toString('utf8').includes('GL_INVALID_OPERATION : glBufferData')) {
-        console.log('Rendering failed due to Electron error \'GL_INVALID_OPERATION: glBufferData\'!'); 
-        if (!attemptedRenderWithoutGPU) {
-          console.log('Retrying rendering with disabled GPU...');
-          attemptedRenderWithoutGPU = true;
-          return spawnRender('--disable-gpu');
-        }
+      // If error is Buffer, print it, otherwise reject
+      if (!!error && error instanceof Buffer) {
+        console.log(di.chalk.yellow(error.toString('utf8')));
+      } else {
+        reject(new Error("Unknown error [" + typeof error + "]: " + error));
       }
+    } 
+
+    // React when rendering process finishes
+    function onClose(code) {
+      if (code !== 0) {
+        reject(new Error("Rendering exited with code " + code));
+      } else {
+        console.log(di.chalk.green('Finished rendering frames in ' + (Date.now() - start) + 'ms.'));
+        resolve();
+      }
```

Comment:
Check if progressBar is complete
```suggestion
      } else {
        if (progressBar.complete) {
          console.log(di.chalk.green('Finished rendering frames in ' + (Date.now() - start) + 'ms.'));
        }
        else console.log(di.chalk.yellow('[render] process completion unverified'));
        resolve();
      }
```

Reply:
but isn't checking for success through UI component a bit strange? shouldn't we prioritize relying on exit codes of process as it's a more robust technique?

Reply:
for instance, the `mergeFrames` function below doesn't check for completion through progress bar, but through finishing all the async merging tasks

Reply:
It is indeed a bit strange but then you realize
1. The `progressBar` is generated by the `frameCount`
https://github.com/faressoft/terminalizer/blob/771548eddea64ee532cb7e8dfceddfdd49337063/commands/render.js#L121
2. The renderer outputs frame indexes to `stdout`
https://github.com/faressoft/terminalizer/blob/771548eddea64ee532cb7e8dfceddfdd49337063/render/index.js#L62-L66
3. The `progressBar` is ticked only when a frame index is caught
https://github.com/faressoft/terminalizer/blob/771548eddea64ee532cb7e8dfceddfdd49337063/commands/render.js#L130-L138
4. Anything could've happened in between that didn't log to `stderr`, yet interrupted the completion of our frame rendering process. Ergo, the need to check if the `progressBar` is complete.

Reply:
Also, the merger doesn't use raw stderr streams to check for errors.
Rather it catches a promise rejection from the `loadPNG()` method which it then forwards to the callback of the `async.eachOfSeries()` method.
This completes the error handling and ensures we don't assume an incomplete merging process as complete.

Reply:
Thanks for the thorough explanation. I was hoping that point 4. would be caught by process exiting.

I'll update this to revert to tracking progress through progressBar after work.